src/tests/e2etest/CommandRunnerTests.cs

```diff
@@ -1,19 +1,279 @@
+        private readonly string _moduleId = "osconfig";
+        private readonly string _deviceId = Environment.GetEnvironmentVariable("E2E_OSCONFIG_DEVICE_ID");
+
+        private readonly string _sasToken = Environment.GetEnvironmentVariable("E2E_OSCONFIG_SAS_TOKEN");
+        private readonly string _uploadUrl = Environment.GetEnvironmentVariable("E2E_OSCONFIG_UPLOAD_URL")?.Trim('\"');
+        private readonly string _resourceGroupName = Environment.GetEnvironmentVariable("E2E_OSCONFIG_RESOURCE_GROUP_NAME")?.Trim('\"');
+
+        public partial class GenericResponse<T>
+        {
+            public T value { get; set; }
+            public int ac { get; set; }
+        }
+
+        [OneTimeSetUp]
+        public void OneTimeSetUp()
+        {
+            _registryManager = RegistryManager.CreateFromConnectionString(_iotHubConnectionString);
+
+            if ((null == _sasToken) || (null == _uploadUrl) || (null == _resourceGroupName))
+            {
+                Assert.Warn("Missing environment vars required for log upload to blob store");
+            }
+        }
+
+        [OneTimeTearDown]
+        public void OneTimeTearDown()
+        {
+            if ((null == _sasToken) || (null == _uploadUrl) || (null == _resourceGroupName))
+            {
+                Console.WriteLine("[TearDown] Uploading logs to blob store");
+
+                string fileName = String.Format("{0}-{1}.tar.gz", _resourceGroupName, _deviceId);
+                string fullURL = String.Format("{0}{1}?{2}", _uploadUrl, fileName, _sasToken);
+
+                string tempFileCommand = "temp_file=`sudo mktemp`";
+                string tarCommand = "sudo tar -cvzf $temp_file osconfig*.log";
+                string curlCommand = String.Format("curl -X PUT -T $temp_file -H \"x-ms-date: $(date -u)\" -H \"x-ms-blob-type: BlockBlob\" \"{1}\"", fileName, fullURL);
+                string uploadCommand = String.Format("cd /var/log && {0} && {1} && {2}", tempFileCommand, tarCommand, curlCommand);
+
+                // Temporarily commented out to avoid uploading logs to blob store twice
+                // if (!ExecuteCommandViaCommandRunner(uploadCommand))
+                // {
+                //     Assert.Warn("[TearDown] Failed to upload logs to blob storage");
+                // }
+            }
+            else
+            {
+                Console.WriteLine("[TearDown] Skipping upload of logs to blob store");
+            }
+        }
+
+        protected bool ExecuteCommandViaCommandRunner(string arguments)
+        {
+            bool success = true;
+            var command = CommandRunnerTests.CreateCommand(arguments);
+
+            try
+            {
+                var desiredResult = SetDesired<CommandRunnerTests.CommandArguments>("CommandRunner", "CommandArguments", command);
+                desiredResult.Wait();
+                int ackCode = desiredResult.Result;
+
+                if (200 != ackCode)
```

Comment:
consider using the `HttpStatusCode::OK` constant - see [OK](https://docs.microsoft.com/en-us/dotnet/api/system.net.httpstatuscode?view=net-6.0#system-net-httpstatuscode-ok)